libtelnet.c

```diff
@@ -1121,6 +1137,47 @@ static void _process(telnet_t *telnet, const char *buffer, size_t size) {
 	}
 }
 
+/* perform NVT end-of-line translation for received text */
+size_t telnet_translate_eol(telnet_t *telnet, char *buffer, size_t size,
+		int *split) {
+	size_t i = 0, l = 0;
+
+	/* no translation if in BINARY mode */
+	if (telnet->flags & TELNET_FLAG_RECEIVE_BINARY)
+		return size;
+
+	/* fix up translation if split across buffer boundary */
+	if (*split && size > 0) {
```

Comment:
both here and below on line ~1171, I think I'd prefer using `i < size` instead of `size > 0`, just because that makes it more directly obvious why the code needs the check at all (because `buffer[i]` is used).